### ISSUE TITLE
about finetune and train

### ISSUE THREAD

**Issue by hdzmtsssw** [redrock303/NeRFLiX_CPVR2023#7]
Hello! I've been working on finetuning the released model with my own dataset, but I seem to be experiencing some issues. I've attempted several approaches, including: 
1. Using `30% my own data(approximately 900 images, 30 per scene, forward-facing) + 70% Vimeo90K` to finetune for `1e5` / `2e5` / `5e4` / `3e5` iterations;
2. Using `30% LLFF-T + 70% Vimeo90K` to finetune for `3e5` iterations;
3. Using `100% my own data` to finetune for `1e4` iterations;
4. Using `25% LLFF-T + 25%Vimeo90K + 50% my own data` to finetune for `1e4` / `6e3` iterations;
5. Using `30% LLFF-T + 70% Vimeo90K` to train from scratch for`2e3` / `1.2e4` iterations.

Despite using different datasets and iterations, all of the finetuned models appear almost identical visually. Although the models appear slightly clearer during testing, they are still significantly worse than the official model.
Furthermore, I attempted to train from scratch, but the results were very similar to the finetuned models. It's possible that I used too few iterations.
I completely followed your instruction and configurations(except`base_lr` 5e-4 -> 1e-4), but I'm wondering if there are any tricks or tips that weren't mentioned in the instruction or paper that could help improve my results. I'm not sure what went wrong, how to train and finetune correctly, or how to obtain a good model.

Here's one of the results:
![00000_finetune](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/58366842/e8919579-a2d9-40d2-885f-01273d863691)

**Comment by redrock303** [comment]
It seems the first experiment will achieve better results since you include as many as 30 additional scenes for fine-tuning. But the enhanced image below still looks noisy, it is strange. How about the enhancement effect on these 30 scenes you just include?
To further improve the result, I suggest not only using the simulated data but also including the real-rendered training views (that have high-quality ground truth images for training). More details can be found in our recent work NeRFLiX++ https://redrock303.github.io/nerflix_plus/.

**Comment by hdzmtsssw** [comment]
Thanks for your reply. Unfortunately, the enhancements on my own scenes are also not good(all the experiments look the same) compared with your released model(more smooth?). I have attempted to train from scratch, but I still obtained noisy results. This is quite confusing.

By the way, do you have any updates on when the code for NeRFLiX++ will be released?

**Comment by redrock303** [comment]
We have not faced such issues. I think you can check/try these points: (1) how sparse your training views are and the overlap between the training views / novel views. (2) may adopt a smaller learning rate for fine-tuning e.g. 5e-6. (3) fixed the encoder and alignment modules only fine-tuning the reconstruction module. (4) check the initial rendering quality of your novel views, as NeRFLiX can not perform accurate correspondence estimation between very noisy novel views and reference views, thus leading to poor enhancement quality.
We will release the source code for NeRFLiX++ upon publication. But, I think you can use a similar idea of NeRFLiX++ to further improve your results by utilizing the training rendered views (with ground truth) as additional paired training samples for finetuning our pre-trained model. The effectiveness is validated by NeRFLiX++ through our recent experiments. Hope these tips may help.

**Comment by hdzmtsssw** [comment]
Hi, @redrock303 
I have just completed the training of the **original version** you released using the **default** settings from scratch, without making any changes except adjusting the `BASE_LR` from 5e-4 to 2e-4 (increasing it beyond that value causes `inf`). However, I did not achieve the same results as the model you provided. Similar to my previous attempts at fine-tuning, the results were almost the same (consistent with the noisy original image, slightly sharper, but significantly less smooth and clear compared to the images obtained from your provided model). I am extremely puzzled by this. 

These issues haven't touched upon the fine-tuning mentioned earlier. Currently, it's impossible for me to replicate your results. Do you have any thoughts on this? Could there be some training details that were not mentioned? Or perhaps, there might be some bugs in the code you've released? Could you please check on this?

![00000_from_scratch](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/58366842/e0eb92fd-6449-4bc7-a562-fe6f94eacc4a)
**Fig1**. Using 30% LLFF-T + 70% Vimeo90K to train from scratch for 3e5 iterations, **all the code and settings were kept as default**, except for the `BASE_LR` which was adjusted from 5e-4 to 2e-4.


![00000_enc_3e5](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/58366842/0ddfe5da-03a2-4b57-92e5-8626e55acd17)
**Fig2**. Using 30% LLFF-T + 70% Vimeo90K to finetune for 3e5 iterations


![00000_enc_llff_3e5](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/58366842/e2d22091-0633-4999-a4e5-2012e39b9544)
**Fig3**. Using 100% LLFF-T to finetune for 3e5 iterations

**Comment by redrock303** [comment]
It is strange that finetuning does not work. In my experience, training on the synthetic dataset (Vimeo90K) alone can achieve significantly better SSIM values (only ~0.13dB improvements in terms of PSNR) than its input views. Recently, I have also finetuned our pre-trained model on some in-the-wild scenes and found that it consistently improves performance.

**Comment by lihui1998** [comment]
> Hi, @redrock303 I have just completed the training of the **original version** you released using the **default** settings from scratch, without making any changes except adjusting the `BASE_LR` from 5e-4 to 2e-4 (increasing it beyond that value causes `inf`). However, I did not achieve the same results as the model you provided. Similar to my previous attempts at fine-tuning, the results were almost the same (consistent with the noisy original image, slightly sharper, but significantly less smooth and clear compared to the images obtained from your provided model). I am extremely puzzled by this.
> 
> These issues haven't touched upon the fine-tuning mentioned earlier. Currently, it's impossible for me to replicate your results. Do you have any thoughts on this? Could there be some training details that were not mentioned? Or perhaps, there might be some bugs in the code you've released? Could you please check on this?
> 
> ![00000_from_scratch](https://user-images.githubusercontent.com/58366842/258687846-e0eb92fd-6449-4bc7-a562-fe6f94eacc4a.png) **Fig1**. Using 30% LLFF-T + 70% Vimeo90K to train from scratch for 3e5 iterations, **all the code and settings were kept as default**, except for the `BASE_LR` which was adjusted from 5e-4 to 2e-4.
> 
> ![00000_enc_3e5](https://user-images.githubusercontent.com/58366842/258687537-0ddfe5da-03a2-4b57-92e5-8626e55acd17.png) **Fig2**. Using 30% LLFF-T + 70% Vimeo90K to finetune for 3e5 iterations
> 
> ![00000_enc_llff_3e5](https://user-images.githubusercontent.com/58366842/258687682-e2d22091-0633-4999-a4e5-2012e39b9544.png) **Fig3**. Using 100% LLFF-T to finetune for 3e5 iterations

Hello, how did you solve this problem?

**Comment by hdzmtsssw** [comment]
@lihui1998 Hi, I apologize for not being able to solve the problem. I have attempted the strategies mentioned above, including fine-tuning with different iterations and training from scratch, but unfortunately, I did not achieve the desired results. If you have any ideas or suggestions, I would be glad to hear them.

**Comment by lihui1998** [comment]
> @lihui1998您好，很抱歉无法解决问题。我尝试了上面提到的策略，包括不同迭代的微调和从头开始训练，但不幸的是，我没有达到预期的结果。如果您有任何想法或建议，我很高兴听到。

I found that the size of the model I trained was different from the model provided by the author.
![image](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/51204863/c1a9da8f-71bd-4fdb-b5f2-b716970f78aa)

**Comment by hitzjx** [comment]
Hello, when I'm reproducing your code, due to resource constraints, I'm using 3 Nvidia RTX 3090 GPUs, setting IMG_PER_GPU to 3, and accordingly setting LR to 2.5e-4. During the training process, there suddenly appears a situation of abnormal grids. Do you know what might be causing this?
![3300_cat_check](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/104314658/ecdabe43-6bde-45bf-a084-5b579f0c387e)
![3350_cat_check](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/104314658/e4fcbf7f-1a90-4fa2-9721-e12830edf834)

**Comment by ZestfulJX** [comment]
Hello, when I'm reproducing your code, due to resource constraints, I'm using 3 Nvidia RTX 3090 GPUs, setting IMG_PER_GPU to 3, and accordingly setting LR to 2.5e-4. During the training process, there suddenly appears a situation of abnormal grids. Do you know what might be causing this?
![3300_cat_check](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/104314658/7f54a2ef-5c2d-44b6-a0bd-20120aa17762)
![3350_cat_check](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/104314658/92bc0eb6-e119-4bc0-9f62-f83975bb00fb)
![微信图片_20240225211601](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/104314658/88667166-f226-432f-9179-2e235db219f8)

**Comment by zouchen92** [comment]
> @lihui1998 Hi, I apologize for not being able to solve the problem. I have attempted the strategies mentioned above, including fine-tuning with different iterations and training from scratch, but unfortunately, I did not achieve the desired results. If you have any ideas or suggestions, I would be glad to hear them.

Hello! I encountered the same problem as you, , including fine-tuning with different iterations and training from scratch, but I did not achieve the desired results.  May I ask if you have resolved this issue now?
